src/lib/statuses.ts

```diff
@@ -29,53 +54,59 @@ export const getStatuses = (
     isHintMode = loaded.isHintMode
   }
 
+  function updateCharObjectKey(letter: string, value: CharStatus) {
+    // Sets a new status of a key, only if the new status is more important than a previous status
+    // i.e. 'present' does not override 'correct', if the character appears twice
+    const currentStatusPriority = charStatusRanking[charObj[toHiragana(letter)]]
+    const newStatusPriority = charStatusRanking[value]
+    if (!currentStatusPriority || newStatusPriority > currentStatusPriority) {
+      charObj[toHiragana(letter)] = value
+    }
+  }
+
   guesses.forEach((word) => {
     unicodeSplit(word).forEach((letter, i) => {
-
       if (isHintMode) {
         vowelStatusKana.forEach((kana) => {
           if (kana.includes(letter) && kana.includes(splitSolution[i])) {
-            //make status close
-            return (charObj[toHiragana(letter)] = 'vowel')
+            updateCharObjectKey(letter, 'vowel')
           }
         })
 
         consonantStatusKana.forEach((kana) => {
           if (kana.includes(letter) && kana.includes(splitSolution[i])) {
-            //make status close
-            return (charObj[toHiragana(letter)] = 'consonant')
+            updateCharObjectKey(letter, 'consonant')
           }
         })
 
         closeStatusKana.forEach((kana) => {
           if (kana.includes(letter) && kana.includes(splitSolution[i])) {
-            //make status close
-            return (charObj[toHiragana(letter)] = 'close')
+            updateCharObjectKey(letter, 'close')
           }
         })
       }
 
-      if (!splitSolution.includes(letter) && !['vowel', 'consonant', 'present', 'close', 'correct'].includes(charObj[toHiragana(letter)])) {
```

Comment:
Previously we had to check that an absent character didn't already have a status. Now that is handled by the priority, so the check here is simpler.